packages/flow-trigger-bundle/src/flow-trigger-display/display.vue

```diff
@@ -0,0 +1,239 @@
+<script setup lang="ts">
+import { ref, unref, computed } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useApi, useStores } from '@directus/extensions-sdk';
+import formatTitle from '@directus/format-title';
+
+const props = withDefaults(
+	defineProps<{
+		value: string;
+		disabled?: boolean;
+		collection: string;
+		field: string;
+		text?: string;
+		icon?: string;
+	}>(),
+	{},
+);
+
+console.log(props);
+
+const { t } = useI18n();
+
+const api = useApi();
+
+const { useFlowsStore, useNotificationsStore } = useStores();
+const flowsStore = useFlowsStore();
+const notificationStore = useNotificationsStore();
+
+const flowMap = computed(() => {
+	const map = new Map<string, any>();
+
+	for (const flow of flowsStore.flows) {
+		map.set(flow.id, flow);
+	}
+
+	return map;
+});
+
+const runningFlows = ref<string[]>([]);
+
+const confirmValues = ref<Record<string, any> | null>();
+const confirmDetails = ref<{
+	description: string;
+	fields: Record<string, any>[];
+} | null>(null);
+
+const isConfirmButtonDisabled = computed(() => {
+	for (const field of confirmDetails.value?.fields || []) {
+		if (
+			field.meta?.required &&
+			(!confirmValues.value ||
+				confirmValues.value[field.field] === null ||
+				confirmValues.value[field.field] === undefined)
+		) {
+			return true;
+		}
+	}
+
+	return false;
+});
+
+const displayCustomConfirmDialog = computed(
+	() => {
+		return Boolean(confirmDetails.value !== null);
+	},
+);
+
+function getConfirmButtonText() {
+	return t('run_flow');
+}
+
+function getButtonText() {
+	if (props.text) {
+		return props.text;
+	}
+
+	const flow = getFlow(props.value);
+	if (flow) {
+		return flow.name;
+	}
+
+	return t('run_flow');
+}
+
+function getButtonIcon() {
+	if (props.icon) {
+		return props.icon;
+	}
+
+	const flow = getFlow(props.value);
+	if (flow) {
+		return flow.icon;
+	}
+
+	return 'bolt';
+}
+
+function confirmRunFlow() {
+	const flow = getFlow(props.value);
+	if (!flow) {
+		return;
+	}
+
+	if (!Boolean(flow.options?.requireConfirmation)) {
+		runFlow();
+	} else {
+		confirmDetails.value = {
+			description: flow.options.confirmationDescription,
+			fields: (flow.options.fields ?? []).map((field: Record<string, any>) => ({
+				...field,
+				name: !field.name && field.field ? formatTitle(field.field) : field.name,
+			})),
+		};
+	}
+}
+
+async function runFlow() {
+	const flow = getFlow(props.value);
+	if (!flow) {
+		return;
+	}
+
+	const flowId = flow.id;
+
+	runningFlows.value = [...runningFlows.value, flowId];
+
+	const collection = props.collection;
+	const keys = [];
```

Comment:
How do I access the primary key of the item related to this display?

Reply:
Unfortunately not possible for displays at this time

Reply:
Per @phazonoverload the display can be removed from the PR for now.

Reply:
I thought the displays got passed the ID when originally writing the spec. I was wrong. Let's remove it for now, budget unchanged, and then we will swing back to it at another point in time as a new piece of work

Reply:
@ComfortablyCoding @phazonoverload Sounds good. Removed the display.